internal/agent/collector/vsphere.go

```diff
@@ -267,49 +292,125 @@ func FillInventoryObjectWithMoreData(vms *[]vspheremodel.VM, inv *apiplanner.Inv
 
 		inv.Vms.PowerStates[vm.PowerState]++
 
-		// Update total values
+		vmDiskTypes := make(map[string]bool)
+
+		for _, disk := range vm.Disks {
+			var diskType string
+
+			if disk.RDM {
+				diskType = "RDM"
+			} else if disk.Datastore.ID != "" {
+				if dsType, exists := datastoreIDToType[disk.Datastore.ID]; exists {
+					diskType = normalizeDiskType(dsType)
+				} else {
+					continue
+				}
+			} else {
+				continue
+			}
+
+			if diskTypeMap[diskType] == nil {
+				diskTypeMap[diskType] = &diskTypeData{
+					vmSet:       make(map[int]struct{}),
```

Comment:
Why do you need to have map here and not just a counter of VMs?
You are in loop of all VMs, so you can know if you have already marked this VM or not
Is there any other usage for this that I don't understand?